server/routes/bookings/bookingDetailsController.ts

```diff
@@ -0,0 +1,48 @@
+import type { RequestHandler } from 'express'
+import { ValidationChain, param, validationResult } from 'express-validator'
+import { VisitService } from '../../services'
+import getPrisonInformation from '../../constants/prisonInformation'
+import paths from '../../constants/paths'
+
+export default class BookingDetailsController {
+  public constructor(private readonly visitService: VisitService) {}
+
+  public view(): RequestHandler {
+    return async (req, res) => {
+      const { booker, bookings } = req.session
+
+      const errors = validationResult(req)
+      if (!errors.isEmpty()) {
+        req.flash('errors', errors.array())
+        return res.redirect(paths.BOOKINGS.HOME)
+      }
+
+      const visitNumber = +req.params.visitNumber
+      const visit = bookings[visitNumber - 1]
+
+      // if manual number entered (larger than current number of visits in session data)
+      // redirect to bookings page
+      if (visitNumber > bookings.length) {
+        return res.redirect(paths.BOOKINGS.HOME)
+      }
```

Comment:
Might want to re-order to check first if visitNumber is going to be out of bounds for the array. E.g. currently it will try to access `bookings[999]` and assign to `visit` **before** checking if this is a valid value.